clients/cli/src/main.rs

```diff
@@ -82,9 +96,97 @@ fn main() -> Result<(), Box<dyn Error>> {
+            );
+            // Check if the wallet address is valid
+            if !keys::is_valid_eth_address(&wallet_address) {
+                let err_msg = format!(
+                    "Invalid Ethereum wallet address: {}. It should be a 42-character hex string starting with '0x'.",
+                    wallet_address
+                );
+                return Err(Box::from(err_msg));
+            }
+            let orchestrator_client = OrchestratorClient::new(environment);
+            let uuid = uuid::Uuid::new_v4().to_string();
+            match orchestrator_client
+                .register_user(&uuid, &wallet_address)
+                .await
+            {
+                Ok(_) => println!("User {} registered successfully.", uuid),
+                Err(e) => {
+                    eprintln!("Failed to register user: {}", e);
+                    return Err(e);
+                }
+            }
+
+            // Save the configuration file with the user ID and wallet address
+            let config = Config::new(
+                uuid,
+                wallet_address,
+                String::new(), // node_id is empty for now
+                environment,
+            );
+            config
+                .save(&config_path)
+                .map_err(|e| format!("Failed to save config: {}", e))?;
+            Ok(())
+        }
+        Command::RegisterNode { env, node_id } => {
+            // Register a new node, or link an existing node to a user.
+            // Requires: a config file with a registered user
+            // If a node_id is provided, update the config with it and use it.
+            // If no node_id is provided, generate a new one.
+            let mut config = Config::load_from_file(&config_path).map_err(|e| {
+                format!("Failed to load config: {}. Please register a user first", e)
+            })?;
+            if config.user_id.is_empty() {
+                return Err(Box::from(
+                    "No user registered. Please register a user first.",
+                ));
+            }
+            if let Some(node_id) = node_id {
+                // If a node_id is provided, update the config with it.
+                println!("Registering node ID: {}", node_id);
+                config.node_id = node_id.to_string();
+                config
+                    .save(&config_path)
+                    .map_err(|e| format!("Failed to save updated config: {}", e))?;
+                println!("Successfully registered node with ID: {}", node_id);
+                Ok(())
+            } else {
+                let environment = env.unwrap_or_default();
+                println!(
+                    "No node ID provided. Registering a new node in environment: {:?}",
+                    environment
+                );
+                let orchestrator_client = OrchestratorClient::new(environment);
+                match orchestrator_client.register_node(&config.user_id).await {
```

Comment:
Similarly the checks above are syntactic and not semantic. Will this call fail if the user id does not exist? Also, is there any concern about the malicious user deliberately registering a node under an id they don't control?